x-pack/plugins/alerting/server/rules_client/lib/increment_revision.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObject } from '@kbn/core/server';
+import { get, isEqual } from 'lodash';
+import { RawRule, RuleTypeParams } from '../../types';
+import { fieldsToExcludeFromRevisionUpdates, UpdateOptions } from '..';
+
+export function incrementRevision<Params extends RuleTypeParams>(
+  currentRule: SavedObject<RawRule>,
+  { data }: UpdateOptions<Params>,
+  updatedParams: RuleTypeParams
+): number {
+  // Diff root level attrs
+  for (const [field, value] of Object.entries(data).filter(([key]) => key !== 'params')) {
+    if (!fieldsToExcludeFromRevisionUpdates.has(field)) {
```

Comment:
Yeah, I agree and tend to avoid them, but it seemed to read easier in this case so I kept them separate. I've got no qualms either way, so just updated it be a single `if` 👍